internal/app/bootkube/main.go

```diff
@@ -80,3 +85,28 @@ func main() {
 		log.Fatalf("bootkube failed: %s", err)
 	}
 }
+
+func init() {
+	// Explicitly set the default http client transport
+	// to work around our fun proxy.Do once bug.
+	// This is the http.DefaultTransport with the Proxy
+	// func overridden so that the environment variables
+	// with be reread/initialized each time the http call
+	// is made.
+	http.DefaultClient.Transport = &http.Transport{
```

Comment:
this shouldn't be required afaik, as `bootkube` runs a service and environment variables don't change over time

Reply:
Ok, maybe I need to add some debug logging then as it did not work for me without this.

Reply:
@salkin what was the error? 

In recent versions, `bootkube` logs are properly handled, so you can pull them with `talosctl logs bootkube`.

Also yesterday another fix went in: #2001, if you are trying to download something from https://

Reply:
ok then it was probably the https issue as yes, trying from https.